syntax = "proto3";

import "engine.proto";
import "google/protobuf/struct.proto"; // For Value

package com.wazuh.api.engine.test;

/******************** Only for test proposes ********************/
message Run_Request
{
    optional string kvdbPath = 1;
    optional string fileStorage = 2;
    optional string policy = 3;
    optional string logLevel = 4;
    optional int32 debugLevel = 5;
    repeated string assetTrace = 6;
    optional bytes protocolQueue = 7;
    optional string protocolLocation = 8;
}

message Run_Response
{
    ReturnStatus status = 1;                  // Status of the query
    optional string error = 2;                // Error message if status is ERROR
    optional google.protobuf.Value value = 3; // Content of the resource if status is OK
}
